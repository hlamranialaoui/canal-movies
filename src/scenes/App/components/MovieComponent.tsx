import React from "react";
import {POSTER_IMAGE_BASE} from "../../../conf/constants";

interface IMovieProps {
    originalName: string;
    posterPath: string;
    voteAverage: number;
}

export default function MovieComponent({originalName, posterPath, voteAverage}: IMovieProps) {
    return (
        <section className="col-lg-3">
            <div className="container">
                <div className="row">
                    <div className="col-lg-12">
                        <img style={{maxWidth: '80%'}} src={`${POSTER_IMAGE_BASE}/${posterPath}`} alt="single"/>
                    </div>
                </div>
                <div className="row align-items-center p-2">
                    <div className="col-lg-12">
                        <h5>{originalName}</h5>
                        <p className="movie-vote-average">
                            <i className="bi bi-star-fill"/> {voteAverage}
                        </p>
                    </div>
                </div>
            </div>
        </section>
    );
}