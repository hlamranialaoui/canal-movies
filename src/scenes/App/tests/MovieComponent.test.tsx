import React from 'react';
import { cleanup, render } from '@testing-library/react';
import MovieComponent from "../components/MovieComponent";

afterEach(cleanup);

describe('renders canal movie component', () => {

    test('renders movie component', async () => {
        const { asFragment } = render(
            <MovieComponent
                originalName={'originalName'}
                posterPath={'posterPath'}
                voteAverage={2}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
