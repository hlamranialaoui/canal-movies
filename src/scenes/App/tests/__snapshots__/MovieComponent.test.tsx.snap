// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders canal movie component renders movie component 1`] = `
<DocumentFragment>
  <section
    class="col-lg-3"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <img
            alt="single"
            src="https://image.tmdb.org/t/p/w500/posterPath"
            style="max-width: 80%;"
          />
        </div>
      </div>
      <div
        class="row align-items-center p-2"
      >
        <div
          class="col-lg-12"
        >
          <h5>
            originalName
          </h5>
          <p
            class="movie-vote-average"
          >
            <i
              class="bi bi-star-fill"
            />
             2
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`renders canal movies app entry renders movies 1`] = `
<DocumentFragment>
  <section
    class="col-lg-3"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <img
            alt="single"
            src="https://image.tmdb.org/t/p/w500/posterPath"
            style="max-width: 80%;"
          />
        </div>
      </div>
      <div
        class="row align-items-center p-2"
      >
        <div
          class="col-lg-12"
        >
          <h5>
            originalName
          </h5>
          <p
            class="movie-vote-average"
          >
            <i
              class="bi bi-star-fill"
            />
             2
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
